src/coin_test/backtest/trade_request.py

```diff
@@ -29,11 +32,134 @@ def __init__(
         """
         self.asset_pair = asset_pair
         self.side = side
-        self.type_ = type_
         self.notional = notional
         self.qty = qty
 
         if notional is not None and qty is not None:
             raise ValueError("Notional and qty cannot be specified together.")
         elif notional is None and qty is None:
             raise ValueError("Must specify either notional or qty")
+
+    @abstractmethod
+    def should_execute(self, price: float) -> bool:
+        """Determine if a trade can execute given the current price.
+
+        Args:
+            price: The current price of the asset
+
+        Returns:
+            bool: True if the trade can execute
+        """
+
+    @abstractmethod
+    def build_trade(self, current_asset_price: dict[AssetPair, pd.DataFrame]) -> Trade:
+        """Build Trade that represents a TradeRequest.
+
+        Args:
+            current_asset_price: Current price data from composer
+
+        Returns:
+            Trade that the TradeRequest represents
+        """
+
+
+class MarketTradeRequest(TradeRequest):
+    """A TradeRequest implementation for market (GTC) orders."""
+
+    def should_execute(self, price: float) -> bool:
+        """A MarketTrade object should always execute."""
+        return True
+
+    def build_trade(self, current_asset_price: dict[AssetPair, pd.DataFrame]) -> Trade:
+        """Build Trade that represents a TradeRequest for a MarketTradeRequest.
+
+        Args:
+            current_asset_price: Current price data from composer
+
+        Returns:
+            Trade that the TradeRequest represents
+        """
+        if self.side == Side.BUY:
+            price = current_asset_price[self.asset_pair]["High"].iloc[0]
+        else:
+            price = current_asset_price[self.asset_pair]["Low"].iloc[0]
+
+        amount = self.qty
+        if amount is None:
+            amount = self.notional // price  # TODO: yeah...
```

Comment:
Maybe not `//`? Don't remember where we landed with this.

Reply:
Nice, we landed  on `/` 
Changed.